File: src/types/apk.ts
export interface ApkVersion {
    version: string
    url: string
    sizeMb?: number
    releaseDate?: string
}

export interface ProgressPayload {
    received: number
    total: number
    percentage: number
}

export interface ExtractionProgress {
    current_file: string
    progress: number
}

File: src-tauri/src/models/apk.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ApkVersion {
    pub version: String,
    pub url: String,
    pub size_mb: Option<f64>,
    pub release_date: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ExtractionProgress {
    pub current_file: String,
    pub progress: f64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct FileEntry {
    pub name: String,
    pub is_dir: bool,
    pub size: u64,
    pub path: String,
}
